### PR TITLE
ADD: Directory Path Display

### PR DESCRIPTION
**Pull request by kosch104**
closes #282

### CODE REVIEW

File: src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java [hunk 1]
```diff
@@ -86,6 +86,20 @@
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @SuppressWarnings({ "PMD.LawOfDemeter" })
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java]:
```suggestion
    @SuppressWarnings(PMD.LawOfDemeter")
```

File: src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java [hunk 2]
```diff
@@ -86,6 +86,20 @@
...
+    public String buildDirectoryPath(long dirId) throws MopsException {
+        String result = "";
+        Directory dir = getDirectory(dirId);
+        if (dir.getParentId() != null) {
+            result = buildDirectoryPath(dir.getParentId()) + "/" + dir.getName();
+        }
+        return result;
+    }
+
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java]: can you make it so the root dir returns "/" and everything else returns something like "/foo/bar/"

File: src/main/java/mops/presentation/DirectoryController.java [hunk 3]
```diff
@@ -72,8 +72,18 @@ public String showFolderContent(RedirectAttributes redirectAttributes,
...
+            dirPath = directoryService.buildDirectoryPath(dirId);
+        } catch (MopsException e) {
+            log.error("Failed to fetch directory path for id '{}':", dirId, e);
+            redirectAttributes.addFlashAttribute("error", new ExceptionPresentationError(e));
+            return "redirect:/material1/error";
+        }
+
         model.addAttribute("fileQueryForm", new FileQueryForm());
         model.addAttribute("account", account);
+        model.addAttribute("directoryPath", dirPath);
```
Comment (iTitus) [inline src/main/java/mops/presentation/DirectoryController.java]: why do you make a new try-catch for this?
just put it in the existing one

**Comment by iTitus** [comment]
Can you add tests for the examples I gave?

File: src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java (hunk 2 above)
Comment (kosch104) [reply src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java]: No - when you return something other than "" the whole material-path redirects to error 404. Debugging not successful, help greatly appreciated

File: src/test/java/mops/businesslogic/directory/DirectoryServiceTest.java [hunk 4]
```diff
@@ -147,6 +147,30 @@ void updatePermissionTest() throws MopsException {
...
+        String shouldReturn = "/a/b";
+        assertThat(directoryService.buildDirectoryPath(createdSecondDir.getId())).isEqualTo(shouldReturn);
+    }
+
+    /**
+     * Test if path display works correctly.
+     */
+    @Test
+    void buildDirectoryPathInRootTest() throws MopsException {
+        String shouldReturn = "";
```
Comment (iTitus) [inline src/test/java/mops/businesslogic/directory/DirectoryServiceTest.java]:
```suggestion
        String shouldReturn = "/";
```

**Comment by realemx** [comment]
Missing correct CSS style for the path, maybe we can get some help by @Thabb after he has finished his work in the styling?

File: src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java [hunk 5]
```diff
@@ -86,6 +86,28 @@
...
+            dir = getDirectory(dir.getParentId());
+        }
+        // add root
+        result.add(dir);
+        List<Directory> resultReversed = new LinkedList<>();
+        //reversing list
+        for (int i = result.size() - 1; i >= 0; i--) {
+            resultReversed.add(result.get(i));
+        }
+        return resultReversed;
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java]: Collections.reverse

**Pull request merged**